### PR TITLE
fix(specs): introduced SourceUpdateDocker

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: -

### Changes included:

To improve clarity and code readability, the 'SourceDocker' was renamed to 'SourceUpdateDocker'. The required property 'configuration' was moved to SourceUpdateDocker, while the 'imageType' property now exists in both the SourceUpdateDocker and SourceDocker. In addition, SourceDocker now includes a reference to SourceUpdateDocker. These changes ensure a more logical and understandable schema structure.

### CODE REVIEW

**Comment by Fluf22** [comment]
Let's see if the `allOf` works as expected 😇

File: specs/ingestion/common/schemas/source.yml [hunk 1]
```diff
@@ -276,11 +276,19 @@ SourceDocker:
...
+SourceDocker:
+  allOf:
+    - $ref: '#/SourceUpdateDocker'
+    - type: object
+      properties:
+        imageType:
+          $ref: '#/DockerImageType'
   required:
     - registry
     - image
```
Comment (shortcuts) [inline specs/ingestion/common/schemas/source.yml:294]: This won't have effect in the allOf, it either needs to be in the update or redefined :/

File: specs/ingestion/common/schemas/source.yml [hunk 2]
```diff
@@ -276,11 +276,19 @@ SourceDocker:
...
+  required:
+    - configuration
+
+SourceDocker:
+  allOf:
+    - $ref: '#/SourceUpdateDocker'
+    - type: object
+      properties:
+        imageType:
+          $ref: '#/DockerImageType'
```
Comment (shortcuts) [inline specs/ingestion/common/schemas/source.yml:291]: Maybe as a standalone object with a define name? so that the generator don't append `AllOf`

**Review by shortcuts — COMMENTED** [review]
noicee

**Comment by Fluf22** [comment]
@shortcuts in the end, I made two different objects so I could control the required list of props.

**Review by shortcuts — APPROVED** [review]
make sense, might be worth to share properties but up to you

**Pull request merged**